alchemyapi.php

```diff
@@ -37,23 +37,26 @@ class AlchemyAPI {
 	  * OUTPUT:
 	  * none
 	*/ 
-	public function AlchemyAPI() {
-		//Load the API Key from api_key.txt
-		$key = trim(file_get_contents("api_key.txt"));
-	
-		if (!$key) {
-			//Keys should not be blank
-			echo 'The api_key.txt file appears to be blank, please copy/paste your API key in the file: api_key.txt', PHP_EOL;
-			echo 'If you do not have an API Key from AlchemyAPI, please register for one at: http://www.alchemyapi.com/api/register.html', PHP_EOL;
-			file_put_contents('api_key.txt','');
-			exit(1);
-		}
+	public function __construct($key = null) {
 
-		if (strlen($key) != 40) {
-			echo strlen($key), PHP_EOL;
-			//Keys should be 40 characters long
-			echo 'It appears that the key in api_key.txt is invalid. Please make sure the file only includes the API key, and it is the correct one.', PHP_EOL;
-			exit(1);
+		if (null === $key) {
+			//Load the API Key from api_key.txt
+			$key = trim(file_get_contents("api_key.txt"));
```

Comment:
Ought to change to 
`$key = trim(file_get_contents(__DIR__."/api_key.txt"));`
in case working directory has changed (such as I've experienced when using with Laravel 5.1).